tests/typescript/conformance/types/intersectionType/intersectionType.ts

```diff
@@ -0,0 +1,6 @@
+type A = & string;
+type B =
+  & { foo: string }
+  & { bar: number };
+
+type C = [& { foo: 1 } & { bar: 2 }, & { foo: 3 } & { bar: 4 }];
```

Comment:
I'm not even sure what the leading `&` does. It seams it's optionally allowed, since it's not in part of the AST. I just copied the test file from TypeScript's repo verbatim.